src/main.cpp

```diff
@@ -299,32 +325,71 @@ int create_csr(const lmp_options &opt, string &key, string &csr)
 	return pkcs11_create_csr(opt, key, csr);
 }
 
+/* Variables shared between main and cleanup_generated_data */
+static lmp_options opt;
+static http_headers headers{{ "Content-type", "application/json" }};
+static bool created_csr = false;
+static bool populated_sota_dir = false;
+
+static void cleanup_generated_data()
+{
+	cout << "Cleaning up partial registration before leaving" << endl;
```

Comment:
A signal may arrive after a device has successfully registered and before all keys/certs are written to a disk.
So, if are removing partially stored stuff for the registered device maybe it makes sense to print some message so a user is aware that a device was actually registered at the backend?

Reply:
We no longer have flags telling at which point the execution was stopped. The logs give some clue, and the user can infer from there if the registration at the backend has succeed. Also, errors are not likely to happen after the registration is successful. Like for https://github.com/foundriesio/lmp-device-register/pull/58#discussion_r1338309645 improvements in the messaging can be part of a separate effort, if we feel is required.